v0/destinations/yahoo_dsp/transform.js

```diff
@@ -0,0 +1,201 @@
+   */
+  switch (AUDIENCE_TYPE[audienceType]) {
+    case "email":
+      // creating the output payload using the audience list and Config
+      dspListPayload = createPayload(traitsList, Config);
+      break;
+    case "deviceId":
+      // throwing error if seedListType is not provided for deviceId type audience
+      if (
+        !seedListType ||
+        (seedListType !== "IDFA" && seedListType !== "GPADVID")
+      ) {
+        throw new CustomError(
+          `[Yahoo_DSP]:: seedListType is required for deviceId type audience and it should be any one of 'IDFA' and 'GPADVID'`,
+          400
+        );
+      }
+      dspListPayload = createPayload(traitsList, Config);
+      dspListPayload = {
+        ...dspListPayload,
+        seedListType
+      };
+      break;
+    case "ipAddress":
+      dspListPayload = createPayload(traitsList, Config);
+      dspListPayload = {
+        ...dspListPayload,
+        seedListType: "SHA256IP"
+      };
+      break;
+    case "mailDomain":
+      // traversing through every element in the add array for the elements to be added.
+      traitsList.forEach(userTraits => {
+        // storing keys of an object inside the add array.
+        const traits = Object.keys(userTraits);
+        // For mailDomain the mailDomain or categoryIds must be present. throwing error if not present.
+        /**
+         * If mailDomain is not provided categoryIds is required. The audience includes consumers who have received
+         * mail from a domain belonging to the specified categories.
+         * Reference for use case of categoryIds:
+         * https://developer.yahooinc.com/dsp/api/docs/traffic/audience/mrt-audience.html#:~:text=Optional-,categoryIds,-Specifies%20an%20array
+         */
+        if (
+          !(
+            traits.includes(AUDIENCE_TYPE[audienceType]) ||
+            traits.includes("categoryIds")
+          )
+        ) {
+          throw new CustomError(
+            `[Yahoo_DSP]:: Required property for ${AUDIENCE_TYPE[audienceType]} type audience is not available in an object`,
+            400
+          );
+        }
+        domains.push(userTraits.mailDomain);
+        categoryIds.push(userTraits.categoryIds);
+      });
+      dspListPayload = { accountId, domains, categoryIds };
+      break;
+    case "pointOfInterest":
+      /**
+       * Here, populating includes and excludes object which contains POI locations visited by the consumer The consumer is
+       * added in the includes if the consumer has visited the POI location and is added in excludes if the
+       * consumer has not visited the POI.
+       */
+      dspListPayload.includes = populateIncludes(traitsList, audienceType);
+      dspListPayload.excludes = populateExcludes(traitsList, audienceType);
+      dspListPayload = { ...dspListPayload, accountId };
+      break;
+    default:
+      throw new CustomError(
+        `[Yahoo_DSP]:: Audience Type "${AUDIENCE_TYPE[audienceType]}" is not supported`,
```

Comment:
no need to address audienceType from AUDIENCE_TYPE as it will be not present in this scenario.

Reply:
done